src/lib.rs

```diff
@@ -427,14 +427,25 @@ pub struct DataFormatDescriptorHeader {
 }
 
 impl DataFormatDescriptorHeader {
-    const LENGTH: usize = 8;
+    pub const LENGTH: usize = 8;
 
     pub const BASIC: Self = Self {
         vendor_id: 0,
         descriptor_type: 0,
         version_number: 2,
     };
 
+    pub fn as_bytes(&self, descriptor_block_size: u16) -> [u8; Self::LENGTH] {
+        let mut output = [0u8; Self::LENGTH];
+
+        let first_word = self.vendor_id | (self.descriptor_type << 17);
+        output[0..4].copy_from_slice(&first_word.to_le_bytes());
+        output[4..6].copy_from_slice(&(self.version_number as u16).to_le_bytes());
+        output[6..8].copy_from_slice(&descriptor_block_size.to_le_bytes());
+
+        output
+    }
+
     fn parse(bytes: &[u8]) -> Result<(Self, usize), ParseError> {
```

Comment:
TODO: Rename to from_bytes? This method returning `(Self, usize)` is weird to me because the length of the DFD block is part of the header.

Reply:
If you can accomplish what needs to happen without this returning a tuple, that (plus a move to from_bytes (or try_from_bytes)) would be fine.

Reply:
Alright, I'll plan on leaving it alone for now.